src/delay.rs

```diff
@@ -0,0 +1,99 @@
+//! Different types of delay for retryable operations.
+
+use std::time::Duration;
+
+use rand::distributions::{IndependentSample, Range as RandRange};
+use rand::thread_rng;
+
+/// Each retry increases the delay since the last exponentially.
+#[derive(Debug)]
+pub struct Exponential {
+    base: u64,
+    current: u64,
+}
+
+impl Exponential {
+    /// Create a new `Exponential` using the given millisecond duration as the initial delay.
+    pub fn from_millis(base: u64) -> Self {
+        Exponential {
+            base: base,
+            current: base,
+        }
+    }
+}
+
+impl Iterator for Exponential {
+    type Item = Duration;
+
+    fn next(&mut self) -> Option<Duration> {
+        let duration = Duration::from_millis(self.current);
+
+        self.current = self.current * self.base;
+
+        Some(duration)
+    }
+}
+
+/// Each retry uses a fixed delay.
+#[derive(Debug)]
+pub struct Fixed {
```

Comment:
With the iterator-based API, this looks just like [`std::iter::repeat`](https://doc.rust-lang.org/std/iter/fn.repeat.html).